client.go

```diff
@@ -939,6 +940,18 @@ func (c *Client) SetContentDecompresserKeys(keys []string) *Client {
 	return c
 }
 
+// SetCircuitBreaker method sets the Circuit Breaker instance into the client.
+// It is used to prevent the client from sending requests that are likely to fail.
+// For Example: To use the default Circuit Breaker:
+//
+//	client.SetCircuitBreaker(NewCircuitBreaker())
+func (c *Client) SetCircuitBreaker(b *CircuitBreaker) *Client {
```

Comment:
This method and the NewCircuitBreaker leads you to have a exported CircuitBreaker, and so the NewCircuitBreaker.

I would like to suggest the option pattern here



```suggestion
func (c *Client) WithCircuitBreaker(opts ...CircuitBreakerOption) *Client {
```

WithCircuitBreaker calling newCircuitBreaker (no longer exporter), newCircuitBreaker returns a circuitBreaker (no longer exporter too)

So the code here is about doing this:

```go
func (c *Client) WithCircuitBreaker(opts ...func(*circuitBreaker)) *Client {
      cb := newCircuitBreaker()
 
      for _, opt := range opts {  
           opt(cb)
      } 

      c.circuitBreaker = cb
}


func WithBreakerTimeout(d time.Duration) func(*circuitBreaker) {
     return func(cb *circuitBreaker) {
         cb.timeout = d
     }
}
```

Reply:
@ccoVeille - It is better to use `SetCircuitBreaker` to enable this feature explicitly.

Reply:
I agree with @jeevatkm on this one, leaving as is